nix/src/main/java/org/gnode/nix/File.java

```diff
@@ -0,0 +1,201 @@
+
+    private native void allocate();
+
+    /**
+     * Copy constructor.
+     *
+     * @param other The file to copy.
+     */
+    public File(@Const @ByRef File other) {
+        allocate(other);
+    }
+
+    private native void allocate(@Const @ByRef File other);
+
+    //--------------------------------------------------
+    // Base class methods
+    //--------------------------------------------------
+
+    private native
+    @Cast("bool")
+    boolean isNone();
+
+    /**
+     * Checks if file is initialized
+     *
+     * @return true if initialized else false
+     */
+    public boolean isInitialized() {
+        return !isNone();
+    }
+
+
+    /**
+     * Opens a file.
+     *
+     * @param name The name/path of the file.
+     * @param mode The open mode. Check {@link FileMode} for options.
+     * @param impl The back-end implementation the should be used to open the file. Check {@link Implementation} for options.
+     * @return The opened file.
+     * @see FileMode
+     * @see Implementation
+     */
+    public static native
+    @ByVal
+    File open(@StdString String name, @Cast("nix::FileMode") int mode,
+              @Cast("nix::Implementation") int impl);
+
+    /**
+     * Opens a file with back-end implementation in hdf5.
+     *
+     * @param name The name/path of the file.
+     * @param mode The open mode. Check {@link FileMode} for options.
+     * @return The opened file.
+     * @see FileMode
+     */
+    public static File open(@StdString String name, int mode) {
+        return open(name, mode, Implementation.Hdf5);
+    }
+
+    /**
+     * Opens a file in ReadWrite mode and with back-end implementation in hdf5.
+     *
+     * @param name The name/path of the file.
+     * @return The opened file.
+     */
+    public static native
+    @ByVal
+    File open(@StdString String name);
+
+    //------------------------------------------------------
+    // Methods for file attribute access.
+    //------------------------------------------------------
+
+    private native
+    @StdVector
+    IntPointer version();
+
+    /**
+     * Read the NIX format version from the file.
+     * <p/>
+     * The version consist of three integers standing for the major, minor and patch version of the nix format.
+     *
+     * @return The format version of the NIX file.
+     */
+    public ArrayList<Integer> getVersion() {
+        return Utils.convertPointerToList(version());
+    }
+
+    /**
+     * Read the format hint from the file.
+     *
+     * @return
+     */
+    public native
+    @Name("format")
+    @StdString
+    String getFormat();
+
+    /**
+     * Return the location / uri.
+     *
+     * @return The uri string.
+     */
+    public native
+    @Name("location")
+    @StdString
+    String getLocation();
+
+    /**
+     * Get the creation date of the file.
+     *
+     * @return The creation date of the file.
+     */
+    public native
+    @Cast("time_t")
+    long createdAt();
```

Comment:
Maybe its a good idea to convert `long` to a more java-esque time representation. My favorite would be Java 8 `LocalDataTime` or `DateTime` from  Joda-Time (http://www.joda.org/joda-time/index.html).